--- libtool-1.4e/ltmain.in.orig	Tue Dec 24 22:53:51 2002
+++ libtool-1.4e/ltmain.in	Tue Dec 24 23:02:09 2002
@@ -2372,7 +2372,7 @@
 	      if test -n "$inst_prefix_dir"; then
 		case "$libdir" in
 		  [\\/]*)
-		    add_dir="-L$inst_prefix_dir$libdir $add_dir"
+		    add_dir="$add_dir -L$inst_prefix_dir$libdir"
 		    ;;
 		esac
 	      fi
--- libtool-1.4e/ltmain.sh.orig	Tue Dec 17 23:52:08 2002
+++ libtool-1.4e/ltmain.sh	Tue Dec 24 23:02:33 2002
@@ -2372,7 +2372,7 @@
 	      if test -n "$inst_prefix_dir"; then
 		case "$libdir" in
 		  [\\/]*)
-		    add_dir="-L$inst_prefix_dir$libdir $add_dir"
+		    add_dir="$add_dir -L$inst_prefix_dir$libdir"
 		    ;;
 		esac
 	      fi
