diff -urN libtool-1.5.2.org/libtool.m4 libtool-1.5.2/libtool.m4
--- libtool-1.5.2.org/libtool.m4	2004-01-25 23:02:49.577134104 +0100
+++ libtool-1.5.2/libtool.m4	2004-01-25 23:09:28.982059578 +0100
@@ -2184,7 +2184,7 @@
 # This must be Linux ELF.
 linux*)
   case $host_cpu in
-  alpha*|hppa*|i*86|ia64*|m68*|mips*|powerpc*|sparc*|s390*|sh*)
+  alpha*|hppa*|i*86|ia64*|m68*|mips*|powerpc*|sparc*|s390*|sh*|x86_64*|amd64*)
     lt_cv_deplibs_check_method=pass_all ;;
   *)
     # glibc up to 2.1.1 does not perform some relocations on ARM
