--- libtool-1.5/libtool.m4.readonlysym	2003-04-24 16:41:59.000000000 -0400
+++ libtool-1.5/libtool.m4	2003-04-24 16:48:59.000000000 -0400
@@ -4384,7 +4384,7 @@
   symcode='[[BCDEGQRST]]'
   ;;
 solaris* | sysv5*)
-  symcode='[[BDT]]'
+  symcode='[[BDRT]]'
   ;;
 sysv4)
   symcode='[[DFNSTU]]'
@@ -4402,7 +4402,7 @@
 # If we're using GNU nm, then use its standard symbol codes.
 case `$NM -V 2>&1` in
 *GNU* | *'with BFD'*)
-  symcode='[[ABCDGISTW]]' ;;
+  symcode='[[ABCDGIRSTW]]' ;;
 esac
 
 # Try without a prefix undercore, then with it.
