--- libtool-1.5/libtool.m4.orig	2003-12-28 15:55:47.000000000 +0100
+++ libtool-1.5/libtool.m4	2003-12-28 16:13:06.571259144 +0100
@@ -2054,7 +2054,7 @@
 [AC_REQUIRE([AC_PROG_EGREP])dnl
 AC_CACHE_CHECK([if the linker ($LD) is GNU ld], lt_cv_prog_gnu_ld,
 [# I'd rather use --version here, but apparently some GNU ld's only accept -v.
-case `"$LD" -v 2>&1 </dev/null` in
+case `$LD -v 2>&1 </dev/null` in
 *GNU* | *'with BFD'*)
   lt_cv_prog_gnu_ld=yes
   ;;
