diff -u libtool-1.5/libtool.m4~ libtool-1.5/libtool.m4
--- libtool-1.5/libtool.m4~	2003-07-08 22:04:05.000000000 +0900
+++ libtool-1.5/libtool.m4	2003-07-08 22:04:05.000000000 +0900
@@ -2744,7 +2744,7 @@
     # linker, instead of GNU ld.  If possible, this setting should
     # overridden to take advantage of the native linker features on
     # the platform it is being used on.
-    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags -o $lib'
+    _LT_AC_TAGVAR(archive_cmds, $1)='$CC -shared -nostdlib $predep_objects $libobjs $deplibs $postdep_objects $compiler_flags -o $lib'
   fi
 
   # Commands to make compiler produce verbose output that lists
