--- libtool-2.4.3/tests/cmdline_wrap.at~	2014-10-27 18:20:51.000000000 +0100
+++ libtool-2.4.3/tests/cmdline_wrap.at	2014-10-30 22:37:17.938397654 +0100
@@ -28,7 +28,7 @@
 AT_SETUP([Run tests with low max_cmd_len])
 AT_KEYWORDS([recursive expensive])
 dnl If we already have failures, then reruns will fail too!
-fail_list=`for f in ?/fail ??/fail ???/fail ????/fail; do echo $f; end`
+fail_list=`for f in ?/fail ??/fail ???/fail ????/fail; do echo $f; done`
 AT_CHECK([test -z "$fail_list" || (exit 77)])
 m4_ifdef([AT_CAPTURE_FILE],
 [AT_CAPTURE_FILE([testsuite.log])])
