diff -urNp -x '*.orig' libtool-2.4.6.org/m4/libtool.m4 libtool-2.4.6/m4/libtool.m4
--- libtool-2.4.6.org/m4/libtool.m4	2015-01-20 17:15:19.000000000 +0100
+++ libtool-2.4.6/m4/libtool.m4	2021-04-09 09:36:10.565189044 +0200
@@ -2127,6 +2127,21 @@ if test no = "$_LT_TAGVAR(lt_cv_prog_com
     AC_MSG_WARN(['$CC' does not support '-c -o', so 'make -j' may be unsafe])
     need_locks=warn
   fi
+  # sanitize
+  new_sys_lib_search_path_spec=
+  for d in $sys_lib_search_path_spec ; do
+    # check if exists
+    if test -d $d -a -x $d ; then
+      # normalize
+      new_d=$(cd $d ; pwd)
+      # uniq
+      if ! echo "$new_sys_lib_search_path_spec" | grep -q -F -e " $new_d " ; then
+        new_sys_lib_search_path_spec="$new_sys_lib_search_path_spec $new_d "
+      fi
+    fi
+  done
+  # compress spaces
+  sys_lib_search_path_spec=`echo "$new_sys_lib_search_path_spec" | $SED -e 's/^ *//g;s/ \+/ /g;s/ $//g'`
 else
   need_locks=no
 fi
