--- libtool-1.4e/libltdl/Makefile.am	Tue Sep 11 01:26:44 2001
+++ libtool-1.4e.new/libltdl/Makefile.am	Mon Mar  3 20:40:35 2003
@@ -46,3 +46,13 @@
 	    || cp $$d/$$file $(DESTDIR)$(datadir)/libtool/libltdl/$$file || :; \
 	  fi; \
 	done
+
+$(srcdir)/Makefile.in:  Makefile.am  $(top_srcdir)/configure.ac $(ACLOCAL_M4)
+	cd $(top_srcdir) && \
+	  $(AUTOMAKE) --foreign  Makefile
+	touch touchfile
+
+touchfile: $(srcdir)/Makefile.in
+
+$(srcdir)/configure:  $(srcdir)/configure.ac $(ACLOCAL_M4) $(CONFIGURE_DEPENDENCIES) touchfile
+	cd $(srcdir) && $(AUTOCONF)
