Index: libtoolize.in
===================================================================
RCS file: /cvsroot/libtool/libtool/libtoolize.in,v
retrieving revision 1.15.2.1
diff -u -r1.15.2.1 libtoolize.in
--- libtoolize.in       28 Nov 2001 00:42:30 -0000      1.15.2.1
+++ libtoolize.in       9 Jan 2002 18:29:57 -0000
@@ -154,7 +154,7 @@
 files='config.guess config.sub ltmain.sh'

 auxdir=.
-auxdirline=`egrep '^AC_CONFIG_AUX_DIR' configure.in 2>/dev/null`
+auxdirline=`egrep '^AC_CONFIG_AUX_DIR' $configure_ac 2>/dev/null`
 if test -n "$auxdirline"; then
   # Handle explicit AC_CONFIG_AUX_DIR settings.
   auxdir=`echo "$auxdirline" | sed 's/^AC_CONFIG_AUX_DIR(\([^)]*\)).*$/\1/'`


